### ISSUE TITLE
Separatist / Coalition Vessel missions (destroying a submarine) is too easy, unless abyss

### ISSUE THREAD

**Issue by JeroonBosch** [Regalis11/Barotrauma#13104]
Separatist / Coalition vessels very often are not in any shape to fight back by the time you encounter them. Monsters absolutely destroy them and sink them, often into the very bottom of the abyss.

Once they're at the bottom of the abyss, they're surviving (crush depth does not kill them because the abyss is too shallow) and so the mission becomes a "kill an abyss creature" mission instead (so it's actually worth the reward then, but not in the intended way).

1. We need to check why AI-controlled vessels are that easily killed. Are bots lacking orders to man the guns? Are there too few gunners? Do they need stronger hulls (do they even have hull upgrades) scaling with difficulty? And then fix that issue
2. We need to make the abyss lethal enough to kill the vessels so it doesn't always turn into an abyss diving mission (separate issue). Abyss monsters seemingly ignore AI vessels?

**Comment by SomeRandomNoobKekeke** [comment]
Actually, the mission becomes: hit them with a nuclear depth charge
And if you make abyss deeper (10k like in dynamic europa) then, after eating a nuke, they will sink to 6k, recover and then slowly float up for an hour (only to eat another nuke)
Abyss creatures seems to ignore them

**Comment by ubertpendragon** [comment]
>     1. We need to check why AI-controlled vessels are that easily killed. Are bots lacking orders to man the guns? Are there too few gunners? Do they need stronger hulls (do they even have hull upgrades) scaling with difficulty? And then fix that issue

AI used to be somewhat bugged but saw multiple improvements https://github.com/Regalis11/Barotrauma/issues/11273 https://github.com/Regalis11/Barotrauma/issues/11272  https://github.com/Regalis11/Barotrauma/issues/11269

Do pirate ships benefit from hull upgrades? 

The biggest offender in PvP would be explosive coilgun ammunition, which shreds through *everything* in the game (everybody agrees on that point), including structures. Don't be afraid to make it less powerful


>     2. We need to make the abyss lethal enough to kill the vessels so it doesn't always turn into an abyss diving mission (separate issue)

We had a situation where the pirate sub sunk to the abyss beyond our own sub's crush depth, so we couldn't kill the pirates

**Comment by Ek-Videogames** [comment]
I think it's mostly some combination of too few gunners & inadequate defensive/support priorities against boarding.

- AI has always been very suboptimal if left to it's own without the agency of a player to help optimize their commands and assist in more difficult tasks, so it's not a surprise to anybody that a crew of "only AIs" would perform this poorly.

- What AI vessels certainly aren't lacking is their spotting ability, able to shoot at the player submarine from beyond actual sonar range. Though sometimes to their detriment as they can tend to waste ammo trying to shoot coilguns at absurd ranges that the projectiles physically can't reach you due to drag and bullet drop.

- AI vessels tend to either get crippled by some large creature, or breached and killed by crawlers/mudraptors. Not sure what to do about the former. For the latter, perhaps an additional security guard not assigned to gunnery instead defending the ship, or allow gunners to more intelligently decide to stop manning the guns and instead begin attacking internal threats if there are no enemies outside and just enemies inside. Or give the other crewmembers somewhat better methods of self defense?

- AI enemy crews don't seem to be capable of medical care? I think I've seen AI pirates self tend for basic injuries, but never give treatment to other injured or downed crewmembers. This makes it essentially impossible for the AI crew to recover from a premature monster attack, or initial contact with the player submarine. Maybe different idle priorities for all crew to allow them give assistance to downed crewmembers they find, and/or the addition of a dedicated medical officer.

- When on the periscopes and on the helm, AI are pretty much sitting ducks and easy to hit with piercing ammunition or a railgun shell, allowing you to fairly easily 'decapitate' the vessel's ability to fight early on making the rest of the engagement fairly easy as you mop up the remaining non-combat personnel. There's not much you can really do about that though since knowing where the gun & command compartments are is simply a case of meta knowledge. Again maybe an AI medic would help, or have AI run away from the periscopes if they start taking repeated damage and don't have any target to attempt to return fire, or some other logic for self-preservation.

**Comment by ubertpendragon** [comment]
To elaborate:

Pirate "ambushes" like `separatistattack1` spawn on top of a potential 2/3 missions which could be monster eliminations. So it's quite likely to end up having pirates and swarms spawn close to eachother, with pirates dying to said swarm.

**Comment by ubertpendragon** [comment]
Another problem is the inefficient suit locker position on both Humpback and Typhon 2 pirate subs

Bots have to run all the way around the sub and back if they don't want to drown and might die before doing anything. Not having a suit easily available and also having to swim through pressure breaks their AI a bit

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/cf1bfd84-f952-4721-8afe-dc93f7211353)
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/407ae95b-75f8-4036-9cf3-5f32a9677396)

**Comment by ubertpendragon** [comment]
2 Giant spinelings spawned on top of the pirate sub, sunk it, the crew scrambled for suits and left the guns, then got swarmed by crawlers and everybody died BEFORE the main sub even reached the pirates

![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/d2a6d5fd-e2da-45ef-a928-5fcd277d1f92)
![image](https://github.com/FakeFishGames/Barotrauma/assets/73229309/48ebb19b-f964-4f45-a47c-dc3368535391)

**Comment by ubertpendragon** [comment]
One more video:

1. Enemies spawn near or on top of the pirate sub. This one barely dodges 3 giant spinelings
2. Bots don't man the guns and get breached without firing a single coilgun shot
3. All of the bots scramble for suits and smgs to fight the first intruder and abandon the guns


Feels like a lot of pirate grievances could be solved by having the enemy sub on autopilot from level end to level start with twice the aggro range.

https://cdn.discordapp.com/attachments/1087504035357147217/1223062034673963059/Desktop_2023.09.10_-_17.38.09.06_37.mp4?ex=66187c48&is=66060748&hm=0568d28eb296f7399dfbfdea7ee9dfd260f42a1fa56bf607e9eec17381b031a6&

**Comment by SomeRandomNoobKekeke** [comment]
MMM, dat vanilla fps

**Comment by Jasontti** [comment]
Tested with unstable changes by running enemy submarine mission multiple times.

Results are mixed. Enemy subs survive more on average than get damaged and were easy to finish off after.
Only once in 10 or so tries they got completely incapacitaded and couldn't recover.

They usually were really close to monster spawn locations and i think ubers suggestion of giving some movement and reducing detection might be a valid way to go.

Didn't have any issues with abyss.
